### ISSUE TITLE
valueInt error on cross-reference filters

### ISSUE THREAD

**Issue by SavvasMohito** [weaviate/weaviate#4027]
## The issue

Hi, I am using python-client v4.4b6 to query a weaviate server v1.23.3. I am facing an issue when trying to apply filters on the amount of cross-references found in objects.

The python query resulting in error:
```python
result = (
    client.collections.get("GTR_project").query.fetch_objects(
        limit=5,
        filters=wvc.Filter(["hasSpinOut"]).greater_than(0),
        return_references=[wvc.QueryReference(link_on="hasSpinOut"), return_properties=["companyName"]],
    ),
)
```

The error message:
*WeaviateQueryException: Query call failed with message extract params: child operand at position 1: Property "hasSpinOut" is a ref prop to the class "GTR_spinOut". Only "valueInt" can be used on a ref prop directly to count the number of refs. Or did you mean to filter on a primitive prop of the referenced class? In this case make sure your path contains 3 elements in the form of [<propName>, <ClassNameOfReferencedClass>, <primitvePropOnClass>].*

Running the exact same query through weaviate console works properly, returning the expected result
```graphql
{
    Get {
      GTR_project (
        where: {
          path: ["hasSpinOut"],
          operator: GreaterThan,
          valueInt: 0
        }
      ) {
        hasSpinOut {
          ... on GTR_spinOut {
             companyName
          }
        }
        _additional {
          id
        }
      }
    }
  }
```

## My thoughts

It seems there is an issue when translating the valueInt filter from python to gql. I tried to debug the error but I could not find where this is happening. I also tried the same filter but with a different cross-reference and still got the same error. Using a valueInt filter on a direct property of the object (not cross-reference) does not cause errors. Can this be another weaviate issue and not a client issue? Thanks in advance.

**Comment by tsmith023** [comment]
This has been fixed and merged into `stable/v1.23` meaning that it will land in the next patch release.

I will leave the issue open until then. For now, you can unblock yourself by using this image in your local environment: `preview-map-correct-dtype-when-filtering-references-by-count-b6ff4ab`

**Comment by SavvasMohito** [comment]
I appreciate your help @tsmith023. Have a nice day!

**Comment by etiennedi** [comment]
Closing as complete based on the statements above.
